micro-rdk/src/esp32/tcp.rs

```diff
@@ -72,7 +72,8 @@ struct Esp32ServerConfig {
 
 impl Esp32ServerConfig {
     fn new(srv_cert: &[u8], srv_key: &[u8]) -> Self {
-        let mut alpn_proto: Vec<_> = vec![ALPN_PROTOCOLS.as_ptr() as *const i8, std::ptr::null()];
+        let mut alpn_proto: Vec<_> =
+            vec![ALPN_PROTOCOLS.as_ptr() as *const c_char, std::ptr::null()];
```

Comment:
I think this is the right fix. On rust 1.83 this wants i8 on xtensa. On rust 1.85 it wants u8 for xtensa. This is due to a change in how they are handling c_char. And I think for riscv on 1.85 it wants something else again.

Reply:
alpn_protos are already stored as c_char so you should be able to just call as_ptr and avoid the casting

Reply:
Discussed in person, we will revisit once we are rust 1.85 or newer everywhere